kedro/extras/extensions/ipython.py

```diff
@@ -83,12 +83,50 @@ def reload_kedro(
         logger.info("Registered line magic '%s'", line_magic.__name__)  # type: ignore
 
 
-def load_ipython_extension(ipython):
-    """Main entry point when %load_ext is executed"""
+def load_ipython_extension(ipython):  # pylint: disable=unused-argument
+    """
+    Main entry point when %load_ext is executed.
+    IPython will look for this function specificially.
+    See https://ipython.readthedocs.io/en/stable/config/extensions/index.html
 
-    global default_project_path
+    This function is called when users do `%load_ext kedro.extras.extensions.ipython`.
+    When user use `kedro jupyter notebook` or `jupyter ipython`, this extension is
+    loaded automatically.
+    """
+    import json
 
-    ipython.register_magic_function(reload_kedro, "line", "reload_kedro")
+    from IPython.core.magic import register_line_magic
+    from IPython.core.magic_arguments import argument, magic_arguments, parse_argstring
+
+    @magic_arguments()
+    @argument(
+        "path", help=("Path to the project root directory."), nargs="?", default=None
```

Comment:
What does `nargs="?"` do?

Reply:
If you `?` on ipython it calls help - super useful